src/SGFTree.cpp

```diff
@@ -401,7 +401,13 @@ std::string SGFTree::state_to_string(GameState& pstate, int compcolor) {
     auto leela_name = std::string{PROGRAM_NAME};
     leela_name.append(" " + std::string(PROGRAM_VERSION));
     if (!cfg_weightsfile.empty()) {
-        leela_name.append(" " + cfg_weightsfile.substr(0, 8));
+        size_t pos = cfg_weightsfile.find_last_of("\\/");
```

Comment:
I remember we had a discussion about using ::basename here (from boost, because it's only in C++17). I'm not sure why that wasn't added, maybe because it requires an additional library to link with and so more project file changes?

The problem with the code above is that it's valid for a Unix path to have a \ in it.

Reply:
I found a similar code solution for windows and according to http://www.cplusplus.com/reference/string/string/find_last_of/ should work. But I guess that it is an easy test to do for anyone that has a windows machine, get my next branch compile leelaz start it and save the sgf file from the console command line and look at the name of the players.

Reply:
The counterexample I gave is on Unix, not Windows.

It's legal to have a weights file named "leela\swa\1.txt".

Maybe we can just ignore that case though :-)